source/docs/network_filter_fuzzing.md

```diff
@@ -0,0 +1,116 @@
+```
+envoy_cc_fuzz_test(
+    name = "network_writefilter_fuzz_test",
+    srcs = ["network_writefilter_fuzz_test.cc"],
+    corpus = "network_writefilter_corpus",
+    # All Envoy network filters must be linked to the test in order for the fuzzer to pick
+    # these up via the NamedNetworkFilterConfigFactory.
+    deps = [
+        ":uber_writefilter_lib",
+        "//source/common/config:utility_lib",
+        "//source/extensions/filters/network/kafka:kafka_broker_config_lib",
+        "//source/extensions/filters/network/mongo_proxy:config",
+        "//source/extensions/filters/network/mysql_proxy:config",
+        "//source/extensions/filters/network/zookeeper_proxy:config",
+        "//source/extensions/filters/network/the_new_filter_created_by_you:config", // <---Add the filter config module here
+        "//test/config:utility_lib",
+    ],
+)
+```
+## Step2. Add the filter name into supported_filter_names
+In [uber_per_writefilter.cc](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/uber_per_writefilter.cc), add the filter name into the vector `supported_filter_names` in method `UberWriteFilterFuzzer::filterNames()`.
+```
+const std::vector<absl::string_view> supported_filter_names = {
+      ...
+      NetworkFilterNames::get().ExtAuthorization, NetworkFilterNames::get().TheNewFilterCreatedByYou,
+      ...
+    };
+```
+
+# Add test cases into corpus
+Good test cases can provide good examples for fuzzers to find more paths in the code, increase the coverage and help find bugs more efficiently.
+Each test case is a file under the folder [network_readfilter_corpus](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/network_readfilter_corpus) or [network_writefilter_corpus](https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/network/common/fuzz/network_writefilter_corpus). It consists of two parts: `config` and `actions`. 
+`config` is the protobuf to instantiate a filter, and `actions` are sequences of actions to take in order to test the filter. 
+An example for testing MongoProxy filter:
+```
+config {
+  name: "envoy.filters.network.mongo_proxy"
+  typed_config {
+    type_url: "type.googleapis.com/envoy.extensions.filters.network.mongo_proxy.v3.MongoProxy"
+    value: "\n\001\\\032\t\032\002\020\010\"\003\010\200t \001"
+  }
+}
+actions {
+  on_new_connection {
+  }
+}
+actions {
+  on_data {
+    data: "\120\0\0\0\1\0\0\0\1\0\0\0\324\7\0\0\4\0\0\0\164\145\163\164\56\164\145\163\164\0\24\0\0\0\377\377\377\377\52\0\0\0\2\163\164\162\151\156\147\137\156\145\145\144\137\145\163\143\0\20\0\0\0\173\42\146\157\157\42\72\40\42\142\141\162\12\42\175\0\0"
+  }
+}
+actions {
+  advance_time {
+    milliseconds: 10000
+  }
+}
+```
+`config.name` is the name of the filter. 
```

Comment:
These are great! It looks like they didn't render with newlines in markdown, maybe add newlines or format as bullets or a table?